src/ompl/geometric/planners/experience/ThunderRetrieveRepair.h

```diff
@@ -0,0 +1,205 @@
+             */
+            const std::vector<ompl::geometric::PathGeometric>& getLastRecalledNearestPaths() const;
+
+            /**
+             *  \brief Get debug information about the top recalled paths that were chosen for further filtering
+             *  \return chosenID - the index of the PlannerData object that was chosen for repair
+             */
+            const std::size_t& getLastRecalledNearestPathChosen() const;
+
+            /**
+             * \brief Get the chosen path used from database for repair
+             * \return PlannerData of chosen path
+             */
+            ompl::geometric::PathGeometric getChosenRecallPath() const;
+
+            /** \brief Get information about the exploration data structure the repair motion planner used each call. */
+            void getRepairPlannerDatas(std::vector<base::PlannerDataPtr> &data) const;
+
+            virtual base::PlannerStatus solve(const base::PlannerTerminationCondition &ptc);
+
+            virtual void clear(void);
+
+            /**
+             * \brief Pass a pointer of the database from the thunder framework
+             */
+            void setExperienceDB(ompl::tools::ThunderDBPtr experienceDB);
+
+            /** \brief Set the planner that will be used for repairing invalid paths recalled from experience */
+            void setRepairPlanner(const base::PlannerPtr &planner);
+
+            virtual void setup(void);
+
+            /**
+             * \brief Repairs a path to be valid in the current planning environment
+             * \param oldPath - from experience
+             * \return true if no error
+             */
+            bool repairPath(ompl::geometric::PathGeometric &path, const base::PlannerTerminationCondition &ptc);
+
+            /**
+             * \brief Use our secondary planner to find a valid path between start and goal, and return that path
+             * \param start - begining state
+             * \param goal - ending state
+             * \param newPathSegment - the solution
+             * \return true if path found
+             */
+            bool replan(const ompl::base::State* start, const ompl::base::State* goal, 
+                        ompl::geometric::PathGeometric &newPathSegment,
+                        const base::PlannerTerminationCondition &ptc);
+          
+            /** \brief Getter for number of 'k' close solutions to choose from database for further filtering */
+            int getNearestK()
+            {
+              return nearestK_;
+            }
+
+            /** \brief Setter for number of 'k' close solutions to choose from database for further filtering */
+            void setNearestK(int nearestK)
+            {
+              nearestK_ = nearestK;
+            }
+
+            /** \brief Helper for outputing a state to console */
+            void debugState(const ompl::base::State* state)
```

Comment:
printState already has std::cout as the default arg. This function does not seem very useful. Could we remove it?